src/components/Drawer.astro

```diff
@@ -41,49 +32,42 @@ const origin = env.siteUrl;
       <!-- Logo -->
       <div class="flex-1 grid justify-items-center lg:justify-items-start">
         <a
-          class={clsx("btn normal-case text-xl logo hover:scale-105", {
-            "bg-secondary hover:bg-secondary": !headerShouldBeDarker,
-            "bg-primary hover:bg-primary": headerShouldBeDarker,
-          })}
+          class="btn normal-case text-white text-xl logo hover:scale-105 bg-secondary hover:bg-secondary"
           href={localizeUrl(`${origin}/`)}
         >
           {t("site-title")}
         </a>
       </div>
       <nav class="flex-none hidden lg:block" role="navigation">
         <ul
-          class={clsx("nav menu menu-horizontal gap-2", {
-            "paint-it-black": headerShouldBeDarker,
-          })}
+          class="nav menu menu-horizontal gap-2"
         >
           <Navbar />
         </ul>
+        <label for="language-selector" class="sr-only">{t("select-label")}</label>
         <LanguageSelector
+          id="language-selector"
           showFlag={false}
-          class={clsx(
-            "select ml-1 select-ghost select-sm font-bold lang-selector focus:text-secondary",
-            {
-              "text-primary": headerShouldBeDarker,
-              "text-secondary": !headerShouldBeDarker,
-            }
-          )}
+          class="select ml-1 select-ghost select-sm font-bold lang-selector focus:text-secondary text-secondary"
         />
       </nav>
     </header>
     <!-- Page content here -->
-    <div role="main">
+    <div role="main" id="content">
       <slot />
     </div>
   </div>
-  <aside class="drawer-side z-50" role="complementary">
-    <label for="drawer-toggle" class="drawer-overlay"></label>
+  <aside class="drawer-side z-50" role="complementary" aria-label={t("drawer-side")}>
+    <label id="drawer-overlay" class="drawer-overlay" aria-label={t("close-menu")}></label>
```

Comment:
questa label è importante che abbia il `for` che punti al giusto input o il toggle del menu non funziona
```suggestion
    <label id="drawer-overlay" for="drawer-toggle" class="drawer-overlay" aria-label={t("close-menu")}></label>
``````